### PR TITLE
#6 add semantic segmentation

### LINKED ISSUE #6: add semantic segmentation
### Search before asking

- [ ] I have searched the [issues](https://github.com///issues) and found no similar feature requests.

### Description

add semantic segmentation

### Use case

_No response_

### Additional

_No response_

### PR DESCRIPTION
**Pull request by yunjunho1**
Semantic segmentation using Unet custum backbone. 
Only the training tools have been added, inference tools will be updated later.

Thank you

### CODE REVIEW

File: autocare_dlt/core/utils/inference.py [hunk 1]
```diff
@@ -281,6 +284,16 @@ def __call__(self, input: torch.Tensor, meta: dict) -> list:
 
         return resized_keypoints.tolist()
 
+# TODO: SegPostProcess for inference(should be project specific?)
```
Comment (siegkeron) [inline autocare_dlt/core/utils/inference.py]: what is different between project specific and general post processing?

File: autocare_dlt/core/utils/inference.py [hunk 2]
```diff
@@ -281,6 +284,16 @@ def __call__(self, input: torch.Tensor, meta: dict) -> list:
...
+class SegPostProcess:
+    def __init__(self, cfg):
+        self.classes = list(cfg.classes)
+
+    def __call__(self, input, meta):
```
Comment (siegkeron) [inline autocare_dlt/core/utils/inference.py:292]: Is it working? or dummy ?

File: autocare_dlt/core/trainer/seg_trainer.py [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+
+        # log
+        if self.iter % 100 == 0 or self.iter == 1:
+            train_mesg = f"epoch {self.epoch} [{self.iter}/{self.iters_per_epoch}] - lr: {self.lr:0.6f} train loss: {self.loss_aver.avg:0.6f} "
+            train_mesg += sup_loss_dict.to_string()
+            if self.rank == 0:
+                logger.info(train_mesg)
+
+        # visualize every first iteration
+        if self.cfg.get("DEBUG_data_sanity", False) and self.iter == 1:
```
Comment (siegkeron) [inline autocare_dlt/core/trainer/seg_trainer.py]: It looks same as what is in det_trainer.

Is it working correctly?

File: autocare_dlt/core/trainer/seg_trainer.py
Comment (siegkeron) [inline autocare_dlt/core/trainer/seg_trainer.py:1]: there is no tblogger, please refer latest DLT main branch and add it

File: autocare_dlt/core/utils/inference.py (hunk 2 above)
Comment (yunjunho1) [reply autocare_dlt/core/utils/inference.py:292]: It's just a dummy. It will be added later

File: autocare_dlt/core/trainer/seg_trainer.py (hunk 3 above)
Comment (yunjunho1) [reply autocare_dlt/core/trainer/seg_trainer.py]: The function to visualize the results of 1 epoch for debug has been removed. A function to visualize the results of segmentation will be added next.

File: autocare_dlt/core/utils/inference.py (hunk 2 above)
Comment (yunjunho1) [reply autocare_dlt/core/utils/inference.py:292]: it has been added

File: autocare_dlt/core/trainer/seg_trainer.py
Comment (yunjunho1) [reply autocare_dlt/core/trainer/seg_trainer.py:1]: Added tblogger to seg trainer

**Review by siegkeron — APPROVED** [review]

**Pull request merged**